### PR TITLE
Make calculateCountForDataOnDates work with unsorted data too

### PR DESCRIPTION
**Pull request by DGiannaris**
### Description:
After a bug on skriba-ink, I realised that the function always expects sorted dates for the data you import. If the data is unsorted the count misses data.

So to amend that I thought about refactoring it to work with both sorted and unsorted data for both modes.

I tested it with live data on ink (locally) to make sure.

We can of course disregard this if we don't like the idea

### CODE REVIEW

File: packages/ui/src/components/Charts/helpers/calculateCountForDataOnDates.ts [hunk 1]
```diff
@@ -29,57 +26,40 @@ export function calculateCountForDataOnDates<T extends Record<string, string[]>>
   // Initialize the indices for each data item
   const indices = {} as Record<StringKeys<T>, number>;
   for (const name in data) {
-    indices[name as StringKeys<T>] = 0;
+    indices[name] = 0;
```
Comment (christosmylonas92) [inline packages/ui/src/components/Charts/helpers/calculateCountForDataOnDates.ts:29]: indices is not being used anymore, safe to remove I guess

Comment (DGiannaris) [reply packages/ui/src/components/Charts/helpers/calculateCountForDataOnDates.ts:29]: True 😅

Comment (DGiannaris) [reply packages/ui/src/components/Charts/helpers/calculateCountForDataOnDates.ts]: [3f326ba](https://github.com/synopsisapp/symbiosis-ui/pull/17/commits/3f326ba8bc621bf9173ff90a24eeb653673adbd6)

File: packages/ui/src/components/Charts/helpers/calculateCountForDataOnDates.test.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import { calculateCountForDataOnDates } from "./calculateCountForDataOnDates";
+
+describe("calculateCountForDataOnDates", () => {
```
[the remark this answers is not in the document]
Comment (DGiannaris) [reply packages/ui/src/components/Charts/helpers/calculateCountForDataOnDates.test.ts:4]: [908f8f5](https://github.com/synopsisapp/symbiosis-ui/pull/17/commits/908f8f5ca05e02e112d96a353173b033fdd4090e)

**Review by bratsos — APPROVED** [review]
Great work! 🎉

**Pull request merged**